java/lance-namespace-hms/src/main/java/com/lancedb/lance/namespace/hive/LanceHMSNamespace.java

```
@@ -0,0 +1,251 @@
+
+    List<String> nss = null;
+    switch (HiveVersion.version()) {
+      case V2:
+        nss = listNamespacesV2(parent);
+        break;
+      case V3:
+        nss = listNamespacesV3(parent);
+        break;
+    }
+
+    Collections.sort(nss);
+    PageUtils.Page page =
+        PageUtils.splitPage(
+            nss, request.getPageToken(), PageUtils.normalizePageSize(request.getPageSize()));
+
+    ListNamespacesResponse response = new ListNamespacesResponse();
+    response.setNamespaces(Sets.newHashSet(page.items()));
+    response.setNextPageToken(page.nextPageToken());
+    return response;
+  }
+
+  private List<String> listNamespacesV2(ObjectIdentifier parent) {
+    try {
+      if (parent.empty()) {
+        return clientPool.run(client -> client.getAllDatabases());
+      } else if (parent.size() <= 2) {
+        return Lists.newArrayList();
+      } else {
+        throw new RuntimeException(String.format("Namespace parent {} is too long.", parent));
+      }
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private List<String> listNamespacesV3(ObjectIdentifier parent) {
+    try {
+      if (parent.empty()) {
+        return clientPool.run(client -> client.getCatalogs());
+      } else if (parent.size() == 1) {
+        return clientPool.run(client -> client.getAllDatabases(parent.level(0)));
+      } else if (parent.size() <= 3) {
+        return Lists.newArrayList();
+      } else {
+        throw new RuntimeException(String.format("Namespace parent {} is too long.", parent));
+      }
+    } catch (TException e) {
+      throw new RuntimeException(e);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public DescribeNamespaceResponse describeNamespace(DescribeNamespaceRequest request) {
+    return null;
```

could you raise UnsupportedOperation instead of returning null?